### PR TITLE
Ability to add items to list

### PR DESCRIPTION
**Pull request by Vinschni**
I added the ability to add items using the filter prompt when pressing the "a" keys.

Furthermore for the textarea i added a function to explicitly set the line number.

### CODE REVIEW

**Comment by bashbunni** [comment]
I think it makes sense to add this feature directly to the list... I'm trying to think of cases where a list might be used, but the program author would want the list to be immutable for the user 🤔

**Comment by muesli** [comment]
While the code is fine, I think prepending a `textinput` can already easily be done as a `list` user. I agree that the convenience would be nice, but we should probably strive for a solution that lets you edit & add items inline (see glow for an example).

**Comment by muesli** [comment]
> I think it makes sense to add this feature directly to the list... I'm trying to think of cases where a list might be used, but the program author would want the list to be immutable for the user thinking

I think there are plenty of such use cases. A fixed list of items isn't uncommon, like retrieving items from an API that doesn't just let you create new ones.

File: list/list.go [hunk 1]
```diff
@@ -878,6 +922,12 @@ func (m *Model) handleFiltering(msg tea.Msg) tea.Cmd {
 			if m.FilterInput.Value() == "" {
 				m.resetFiltering()
 			}
+		case key.Matches(msg, m.KeyMap.AcceptWhileAdding):
+			m.hideStatusMessage()
+			m.FilterInput.Blur()
+			m.filterState = Unfiltered
+			m.updateKeybindings()
+			return func() tea.Msg { return ItemAddedMsg(m.FilterInput.Value()) }
```
Comment (meowgorithm) [inline list/list.go:930]: Small note here: logically speaking, there's no need to send a message for internal app communication. That is, commands are strictly for IO.

**Comment by bashbunni** [comment]
@muesli 
Just saw this PR again, here are my thoughts. It sounds like we're leaning away from implementing this based on our discussion above. Should I close the ticket?

**Comment by maaslalani** [comment]
Hey @Vinschni, thanks so much for this contribution! ❤️ 

I'm cleaning up some old PRs and it looks like we want a more generic / flexible solution for adding items. If you still need this feature please feel free to re-open this PR or create a new issue we can tackle!

**Pull request closed without merging**